perllib/FixMyStreet/Cobrand/IsleOfWight.pm

```diff
@@ -70,6 +70,92 @@ sub open311_munge_update_params {
     $params->{description} = "FMS-Update: " . $params->{description};
 }
 
+sub munge_category_list {
+    my ($self, $options, $contacts, $extras) = @_;
+
+    my $user = $self->{c}->user;
+    my %bodies = map { $_->body->name => $_->body } @$contacts;
+    my $b = $bodies{'Isle of Wight Council'};
+
+    if ( $user && ( $user->is_superuser || $user->belongs_to_body( $b->id ) ) ) {
+        return;
+    }
+
+    @$contacts = grep { $_->send_method && $_->send_method eq 'Triage' } @$contacts;
+    my $seen = { map { $_->category => 1 } @$contacts };
+    @$options = grep { my $c = ($_->{category} || $_->category); $c =~ 'Pick a category' || $seen->{ $c } } @$options;
+}
+
+sub munge_category_where {
+    my ($self, $where) = @_;
+
+    my $user = $self->{c}->user;
+    my $b = $self->{c}->model('DB::Body')->for_areas( $self->council_area_id )->first;
+    if ( $user && ( $user->is_superuser || $user->belongs_to_body( $b->id ) ) ) {
+        $where = {
+            body_id => $where->{body_id},
+            -or => { send_method => [ { '!=' => 'Triage' }, undef ] },
+        };
+        return $where;
+    }
+
+    $where->{'send_method'} = 'Triage';
+    return $where;
+}
+
+sub munge_load_and_group_problems {
+    my ($self, $where, $filter) = @_;
+
+    return unless $where->{category};
+
+    my @cat_names = $self->expand_triage_cat_list($where->{category});
+
+    $where->{category} = \@cat_names;
+    my $problems = $self->problems->search($where, $filter);
+    return $problems;
+}
+
+sub munge_filter_category {
+    my $self = shift;
+
+    my $c = $self->{c};
+    return unless $c->stash->{filter_category};
+
+    my @cat_names = $self->expand_triage_cat_list([ keys %{$c->stash->{filter_category}} ]);
+    $c->stash->{filter_category} = { map { $_ => 1 } @cat_names };
+}
+
+sub expand_triage_cat_list {
+    my ($self, $categories) = @_;
+
+    my $b = $self->{c}->model('DB::Body')->for_areas( $self->council_area_id )->first;
+
+    my $cats = $self->{c}->model('DB::Contact')->not_deleted->search(
+        {
+            body_id => $b->id,
+            category => $categories
+        }
+    );
+
+    my @cat_names;
+    while ( my $cat = $cats->next ) {
+        if ( $cat->send_method eq 'Triage' ) {
+            my $cat_name = $cat->category;
+            my $len = length $cat_name;
+            my @cats = $self->{c}->model('DB::Contact')->not_deleted->search(
```

Comment:
As there won't be many, but this loop might be a few times, wonder if it's easier/nicer code to fetch all the body's contacts first off, and create a lookup of group => categories, then you don't have to worry about RABX string lengths at least!